impeller/entity/shaders/glyph_atlas_sdf.frag

```diff
@@ -0,0 +1,34 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+uniform sampler2D glyph_atlas_sampler;
+
+uniform FragInfo {
+  vec2 atlas_size;
+  vec4 text_color;
+} frag_info;
+
+in vec2 v_unit_vertex;
+in vec2 v_atlas_position;
+in vec2 v_atlas_glyph_size;
+
+out vec4 frag_color;
+
+void main() {
+  vec2 scale_perspective = v_atlas_glyph_size / frag_info.atlas_size;
+  vec2 offset = v_atlas_position / frag_info.atlas_size;
+
+  // Inspired by Metal by Example's SDF text rendering shader:
+  // https://github.com/metal-by-example/sample-code/blob/master/objc/12-TextRendering/TextRendering/Shaders.metal
+
+  // Outline of glyph is the isocontour with value 50%
+  float edge_distance = 0.5;
+  // Sample the signed-distance field to find distance from this fragment to the glyph outline
+  float sample_distance = texture(glyph_atlas_sampler, v_unit_vertex * scale_perspective + offset).a;
+  // // Use local automatic gradients to find anti-aliased anisotropic edge width, cf. Gustavson 2012
+  float edge_width = length(vec2(dFdx(sample_distance), dFdy(sample_distance)));
+  // // Smooth the glyph edge by interpolating across the boundary in a band with the width determined above
+  float insideness = smoothstep(edge_distance - edge_width, edge_distance + edge_width, sample_distance);
```

Comment:
How are you planning to address non-antialiased drawing? Given the template solution, It seems like binding an extra float uniform for all atlas implementations wouldn't be so bad even if unused in some of them.

Reply:
Yeah. I'm not married to the template solution, but it saved me from copy/pasting a bunch of code that was nearly identical. There's already a branch in there based on the type of glyph atlas, we could add another branch for SDF if needed.

Reply:
Ahh except that branch doesn't work lol.

Reply:
Ok, fixed the branching using `std::is_same`, and we should be able to do that for other parameters we need here ... although at some point it may make sense to stop using templates and just replicate some code or create additional inline helper functions or something.

Reply:
The actual metal by example drawing seemed to handling some AA here in a way that didn't quite work, it'd probably be worth revisiting.